crates/core/src/client/config.rs

```diff
@@ -22,6 +22,25 @@ pub enum Network {
     ProviderUrl(Url),
 }
 
+impl Network {
+    pub fn get_gateway_url(&self) -> Result<Url, ConfigError> {
```

Comment:
let's use as little as possible the word "get" for getters in rust

https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter